packages/common/src/types/semanticLayer.ts

```diff
@@ -149,9 +149,80 @@ export type SemanticLayerQueryPayload = {
     context: 'semanticViewer';
 };
 
-export enum SemanticLayerStringFilterOperator {
+// Helper functions and constants
+const SEMANTIC_LAYER_DEFAULT_QUERY_LIMIT = 500;
+
+export const isSemanticLayerTimeDimension = (
+    field: SemanticLayerField,
+): field is SemanticLayerTimeDimension => 'granularity' in field;
+
+export function getDefaultedLimit(
+    maxQueryLimit: number,
+    queryLimit?: number,
+): number {
+    return Math.min(
+        queryLimit ?? SEMANTIC_LAYER_DEFAULT_QUERY_LIMIT,
+        maxQueryLimit,
+    );
+}
+
+// Semantic Layer Scheduler Job
+
+export const semanticLayerQueryJob = 'semanticLayer';
+
+export type SemanticLayerJobStatusSuccessDetails = {
+    fileUrl: string;
+    columns: string[];
+};
+
+export type SemanticLayerJobStatusErrorDetails = {
+    error: string;
+    charNumber?: number;
+    lineNumber?: number;
+    createdByUserUuid: string;
+};
+
+export type ApiSemanticLayerJobStatusResponse = {
+    status: 'ok';
+    results: {
+        status: SchedulerJobStatus;
+        details:
+            | SemanticLayerJobStatusSuccessDetails
+            | SemanticLayerJobStatusErrorDetails;
+    };
```

Comment:
by any chance if you want to make this "stronger"

since this type

```ts
enum SchedulerJobStatus {
    SCHEDULED = 'scheduled',
    STARTED = 'started',
    COMPLETED = 'completed',
    ERROR = 'error',
}
```

it can be this:

```ts
results:
        | {
              status: SchedulerJobStatus.SCHEDULED;
              details?: undefined;
          }
        | {
              status: SchedulerJobStatus.STARTED;
              details?: undefined;
          }
        | {
              status: SchedulerJobStatus.COMPLETED;
              details: SemanticLayerJobStatusSuccessDetails;
          }
        | {
              status: SchedulerJobStatus.ERROR;
              details: SemanticLayerJobStatusErrorDetails;
          };
```